hdf/src/hfiledd.c

```diff
@@ -328,7 +326,7 @@ HTPinit(filerec_t *file_rec, /* IN: File record to store info in */
         ndds = MIN_NDDS;
 
     /* allocate the dd block in memory and initialize it */
-    file_rec->ddhead = (ddblock_t *)HDmalloc(sizeof(ddblock_t));
+    file_rec->ddhead = (ddblock_t *)malloc(sizeof(ddblock_t));
     if (file_rec->ddhead == (ddblock_t *)NULL)
```

Comment:
Is cast necessary

Reply:
I haven't looked but yeah, EVERYTHING was casted, many of which I had no idea why...